apisix/plugins/error-log-logger.lua

```diff
@@ -0,0 +1,224 @@
+        return res, err_msg
+    end
+
+    sock:settimeout(config.timeout * 1000)
+
+    local ok, err = sock:connect(config.host, config.port)
+    if not ok then
+        err_msg = "failed to connect the TCP server: host[" .. config.host
+                  .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        return res, err_msg
+    end
+
+    if config.tls then
+        ok, err = sock:sslhandshake(true, config.tls_options, false)
+        if not ok then
+            return false, "failed to to perform TLS handshake to TCP server: host["
+                          .. config.host .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        end
+    end
+
+    table.insert(data, "\n")
+    local bytes, err = sock:send(data)
+    if not bytes then
+        sock:close()
+        err_msg = "failed to send data to TCP server: host[" .. config.host
+                  .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        return res, err_msg
+    end
+
+    sock:setkeepalive(config.keepalive * 1000)
+    return true
+end
+
+
+local function process()
+    if not loaded_flag then
+        local code, body = plugin_metadata.get(plugin_name)
+        if code == 200 then
+            config = body.node.value
+        else
+            core.log.info("there is no config for ", plugin_name, ", use the default config")
+        end
+
+        if not check_metadata_schema(config) then
+            core.log.info("check metadata failed, the config:", core.json.delay_encode(config))
+            return
+        end
+
+        local level = log_level[string.upper(config.level)]
+        local status, err = errlog.set_filter_level(level)
+        if not status then
+            core.log.warn("failed to set filter level by ngx.errlog, the error is :", err)
+            return
+        else
+            core.log.info("set the filter_level to ", config.level)
+        end
+
+        loaded_flag = true
+    end
+
+    local id = ngx.worker.id()
+    local entries = {}
+    local logs = errlog.get_logs(10)
+    while ( logs and #logs>0 ) do
+        for i = 1, #logs, 3 do
+            table.insert(entries, logs[i + 2])
+        end
+        logs = errlog.get_logs(10)
+    end
+
+    if #entries == 0 then
+        return
+    end
+
+    local log_buffer = buffers[id]
+    if log_buffer then
+        for i = 1, #entries do
+            log_buffer:push(entries[i])
+        end
+        return
+    end
+
+    -- Generate a function to be executed by the batch processor
+    local func = function(entries)
+        return send_to_server(entries)
+    end
+    local config_bat = {
+        name = config.name,
+        retry_delay = config.retry_delay,
+        batch_max_size = config.batch_max_size,
+        max_retry_count = config.max_retry_count,
+        buffer_duration = config.buffer_duration,
+        inactive_timeout = config.inactive_timeout,
+    }
+
+    local err
+    log_buffer, err = batch_processor:new(func, config_bat)
```

Comment:
we can use `send_to_server` directly, all right? then we can remove https://github.com/apache/apisix/pull/2488/files#diff-f33917797f0ad1ad27d82b581c7ba6afbd62abed8f6fc3038d9663ad0bc0519cR175-R177